Cli for Microsoft 365 release blog post v10 (#1961)
pnp/blog: content/cli-for-microsoft-365/cli-for-microsoft-365-v10-0/index.md

---

> title: CLI for Microsoft 365 v9

Reviewer: ```suggestion
title: CLI for Microsoft 365 v10
```

date: 2024-10-31T01:00:00.000Z
author: Adam Wójcik
githubname: Adam-it
categories:
  - CLI for Microsoft 365
images:
  - images/banner-cli-m365.png
tags:
  - CLI for Microsoft 365
  - Entra ID
  - Viva
  - Viva Engage
  - SharePoint
  - SharePoint Premium
type: popular
---

> We have just published a new major version of CLI for Microsoft 365 v10. [CLI for Microsoft 365](https://aka.ms/cli-m365) is a cross-platform command-line tool that allows you to manage your Microsoft 365 tenant and SharePoint Framework projects. This major release introduces several new commands that will enhance the quality of your Microsoft 365 experience as well as a breaking change in the default login experience.

Reviewer 2: We mention here only one breaking change while later on there are several. I suggest we broaden the statement in the intro paragraph to more clearly communicate that there are several breaking changes.

Author: ajj a copy paste mistake from previous major. sorry for that

We know, another major release! But this one is not an ASAP reaction to a breaking change we needed to adapt to. This one is is the longly awaited and planned release that will introduce a lot of helpful improvements to our product. The [list of breaking changes](https://pnp.github.io/cli-microsoft365/about/release-notes#%EF%B8%8F-breaking-changes) is quite long but it introduces a lot of helpful and necessary improvements to CLI for Microsoft 365 and also aligns it with the the latest changes and naming updates of Microsoft 365 products and apps. 

A good example of necessary changes that were done in order to align with Microsoft 365 were 
- [removal of the `AAD` alias](https://pnp.github.io/cli-microsoft365/v10-upgrade-guidance#removed-aad-options-and-aliasses-to-aad-commands), which means from now on we will be using only the `Entra` word when referring to Entra ID
- We also [removed aliases to good old Yammer and therefore leaving only Viva Engage](https://pnp.github.io/cli-microsoft365/v10-upgrade-guidance#removed-aad-options-and-aliasses-to-aad-commands) to align all our commands with the current state of Microsoft 365

> We also introduced breaking changes that will greatly improve some functionalities of our product. For example, we updated the API endpoints that were used in [spo folder move](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-move/) and [spo folder copy](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-copy/) commands so that now it will be possible to manage large folders between site collections. As now we use different APIs we also need to adapt our command options to what is supported in the new endpoints. You may read more details about this change here: 

Reviewer 3: ```suggestion
We also introduced breaking changes that will greatly improve some functionalities of our product. For example, we updated the API endpoints that were used in [spo folder move](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-move/) and [spo folder copy](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-copy/) commands so that now it will be possible to copy/move larger folders between site collections. As we now use different APIs we also need to adapt our command options to what is supported in the new endpoints. You may read more details about this change here: 
```

> We also introduced breaking changes that will greatly improve some functionalities of our product. For example, we updated the API endpoints that were used in [spo folder move](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-move/) and [spo folder copy](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-copy/) commands so that now it will be possible to manage large folders between site collections. As now we use different API we also need to adapt our command options to what is supported in the new endpoints. You may read more details about this change here: 

Reviewer: ```suggestion
We also introduced breaking changes that will greatly improve some functionalities of our product. For example, we updated the API endpoints that were used in [spo folder move](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-move/) and [spo folder copy](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-copy/) commands so that now it will be possible to manage large folders between site collections. As now we use different APIs we also need to adapt our command options to what is supported in the new endpoints. You may read more details about this change here: 
```

- [Updated command spo folder move and spo folder copy](https://pnp.github.io/cli-microsoft365/v10-upgrade-guidance#updated-command-spo-folder-move-and-spo-folder-copy)

> Similar improvements we introduced to [spo file copy](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-copy/) and [spo file move](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-move/) commands. We also updated those commands to new endpoints to ensure the command's functionality and reliability. You may find more details about those changes here: 

Reviewer 3: ```suggestion
Similar improvements are introduced to [spo file copy](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-copy/) and [spo file move](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-move/) commands. We also updated those commands to new endpoints to ensure the command's functionality and reliability. You may find more details about those changes here: 
```

- [Updated command spo file copy](https://pnp.github.io/cli-microsoft365/v10-upgrade-guidance#updated-command-spo-file-copy)
- [Updated command spo file move](https://pnp.github.io/cli-microsoft365/v10-upgrade-guidance#updated-command-spo-file-move)

Other breaking changes were introduced to align some of our commands with our conventions so that we are consistent and follow similar approaches in all places. For example we: 
- added `--force` option to [spo site appcatalog remove](https://pnp.github.io/cli-microsoft365/cmd/spo/site/site-appcatalog-remove)

> - or we updated the purpose of option `displayName` for [entra m365group set command](https://pnp.github.io/cli-microsoft365/cmd/entra/m365group/m365group-set) 

Reviewer 3: ```suggestion
- updated the purpose of option `displayName` for [entra m365group set command](https://pnp.github.io/cli-microsoft365/cmd/entra/m365group/m365group-set) 
```

This is just the tip of the iceberg of the necessary and longly awaited updates we introduced to CLI for Microsoft 365 that will greatly improve its usage and align it with Microsoft 365!

> To help you go over all the changes and upgrade to v10, we prepared a summary of the breaking changes and the recommended actions. For more information see the [v10 Upgrade Guidance](https://pnp.github.io/cli-microsoft365/v10-upgrade-guidance) on our docs.

> For the complete list of what's new and changed, see the [release notes](https://pnp.github.io/cli-microsoft365/about/release-notes#v1000).

## What else is new

### New SharePoint Premium commands

> This major release also introduce a brand new [spp model list](https://pnp.github.io/cli-microsoft365/cmd/spp/model/model-list/) command which will allow you to retrieve the list of unstructured document processing models.

Reviewer: ```suggestion
This major release also introduces a brand new [spp model list](https://pnp.github.io/cli-microsoft365/cmd/spp/model/model-list/) command which will allow you to retrieve the list of unstructured document processing models.
```

> The usage is as always very simple and in order to retrieve a list of SharePoint Premium unstructured document processing models on the content center site, simply run:

Reviewer 2: Avoid repeating "simple" in one sentence.

Author: done

```sh
m365 spp model list --siteUrl https://contoso.sharepoint.com/sites/ContentCenter
```

For more information check the following resources:
- [m365 spp model list](https://pnp.github.io/cli-microsoft365/cmd/spp/model/model-list/)

### New Viva engage command

We don't stop enhancing the Viva Engage area with new commands that will help you automate and manage this part of Microsoft 365. In this release, we added a command that will allow you to get all users within a specified Microsoft 365 Viva Engage community.

To list all admins from a community specified by display name run:

```sh
m365 viva engage community user list --communityDisplayName "All company" --role Admin
```

> or you may get the list of members by specifying the group ID by running:

Reviewer: ```suggestion
Or you may get the list of members by specifying the group ID by running:
```

```

Or to update the expiration date-time of an anonymous sharing link for a folder specific by folder URL, run: 

```sh
m365 spo folder sharinglink set --webUrl https://contoso.sharepoint.com/sites/demo --folderUrl /sites/demo/shared%20documents/Folder --id 7c9f97c9-1bda-433c-9364-bb83e81771ee --expirationDateTime '2022-11-30T00:00:00Z'
```

For more information check the following resources:
- [m365 spo folder sharinglink set](https://pnp.github.io/cli-microsoft365/cmd/spo/folder/folder-sharinglink-set/)

### New SharePoint admin management command

In this release, we also introduced a new command that allows you to retrieve the information about default site group membership.

It is as easy as:

```sh
m365 spo tenant site membership list --siteUrl https://contoso.sharepoint.com
```

For more information check the following resources:
- [m365 spo tenant site membership list](https://pnp.github.io/cli-microsoft365/cmd/spo/tenant/tenant-site-membership-list/)

## What's changed

> Besides all these new commands, we've also made some changes to some existing commands. A few bugs have been fixed and the codebase has been polished. Let's have a closer look at some of the updates that stand out the most

Reviewer: ```suggestion
Besides all these new commands, we've also made some changes to some existing commands. A few bugs have been fixed and the codebase has been polished. Let's have a closer look at some of the updates that stand out the most.
```

### Extended `login` command with `--ensure` option

We updated our `login` command adding a new option that ensures that the user is signed in. If the user isn't signed in, it initiates the login flow. This option will allow you to update and simplify your scripts as now you will no longer need to run the `m365 status` command first to ensure that you are logged in and if not run the `login` command which is typically the first thing we do in almost every script. Now you may do all that by simply running `m365 login --ensure`.

- [m365 login](https://pnp.github.io/cli-microsoft365/cmd/login/)

### Enhanced the `spo list remove` command with `--recycle` option

We improved the [spo list remove](https://pnp.github.io/cli-microsoft365/cmd/spo/list/list-remove) with a new `--recycle` option that when used will send the list to the recycle bin instead of permanently deleting it.

### Renamed the default CLI for Microsoft 365 app registration created during `m365 setup`

> Not so long ago we updated the default behavior of the login command which now requires you to provide an app registration which will allow CLI for Microsoft 365 to sign in to your tenant. You may read more details about this change in [previous major release blog post](https://pnp.github.io/blog/cli-for-microsoft-365/cli-for-microsoft-365-v9-0/#the-new-major-version-of-cli-for-microsoft-365--v9). Along with this change we updated the `m365 setup` command so that it now also allows you to create the required app registration on your tenant along with all required settings and selected set of permissions. In this release we updated the default name of this app to 'CLI for M365', previously it was 'CLI for Microsoft 365'. Unfortunately, it turned out that the word 'Microsoft' in your app registration name blocks it from becoming multitenant. That is why in order to avoid this unnecessary problem we updated the name.

Author: TBH I would. Creating this app reg is a brand new thing we introduced and already has a change/update so I guess it's worth being transparent here as well

> Not so long ago we updated the default behavior of the login command which now requires you to provide an app registration which will allow CLI for Microsoft 365 to sign in to your tenant. You may read more details about this change in [previous major release blog post](https://pnp.github.io/blog/cli-for-microsoft-365/cli-for-microsoft-365-v9-0/#the-new-major-version-of-cli-for-microsoft-365--v9). Along with this change we update the `m365 setup` command so that it now also allows you to create the required app registration on your tenant along with all required settings and selected set of permissions. In this release we updated the default name of this app to 'CLI for M365', previously it was 'CLI for Microsoft 365'. Unfortunately, it turned out that the word 'Microsoft' in your app registration name blocks it from becoming multitenant. That is why in order to avoid this unnecessary problem we updated the name.

Reviewer: ```suggestion
Not so long ago we updated the default behavior of the login command which now requires you to provide an app registration which will allow CLI for Microsoft 365 to sign in to your tenant. You may read more details about this change in [previous major release blog post](https://pnp.github.io/blog/cli-for-microsoft-365/cli-for-microsoft-365-v9-0/#the-new-major-version-of-cli-for-microsoft-365--v9). Along with this change we updated the `m365 setup` command so that it now also allows you to create the required app registration on your tenant along with all required settings and selected set of permissions. In this release we updated the default name of this app to 'CLI for M365', previously it was 'CLI for Microsoft 365'. Unfortunately, it turned out that the word 'Microsoft' in your app registration name blocks it from becoming multitenant. That is why in order to avoid this unnecessary problem we updated the name.
```
